nextstrain/cli/command/setup.py

```diff
@@ -70,9 +70,9 @@ def run(opts: Options) -> int:
     print(heading(f"Checking setup…"))
 
     if not opts.dry_run:
-        tests = opts.runner.test_setup()
+        tests: RunnerTestResults = opts.runner.test_setup()
 
-        print_runner_tests(tests)
+        tests = stream_and_print_results(tests)
 
         if not runner_tests_ok(tests):
```

Comment:
_non-blocking_

This API is a little weird. I'd wonder about something more like 

```python
ok = print_and_check_runner_tests(tests)

if ok:
```

by having `print_and_check_runner_tests()` buffer internally and `return runner_tests_ok(tests)`.

Reply:
Updated with 0597de130942e1a66b238b992def830d660e3083